cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -589,6 +590,23 @@ public int addApplication(ApplicationId id, ApplicationMeta meta) throws Conflic
     }, ConflictException.class);
   }
 
+  @Override
+  public void updateApplicationSourceControlMeta(List<UpdateSourceControlMetaRequest> updateRequests) throws IOException {
```

Comment:
I don't understand what the intent of this method is. As implemented, it looks like it is trying to create a new application version with the current app spec but with a different file hash. But why would we want to do that? The file hash is tied to the application information, if it is changing the application should also be changing.

Reply:
I have updated the implementation to update only the source_control_metadata column.